api/history.go

```diff
@@ -29,18 +40,59 @@ import (
 //       404: errorResp
 //       500: errorResp
 var historyHandler handler = func(w http.ResponseWriter, r *http.Request) *errorResponse {
+	follow, err := getBoolParam(r.URL, "follow")
+	if err != nil {
+		return err
+	}
+
+	var enc *json.Encoder
+	if follow {
+		// Ensure every write is a flush.
+		f, ok := w.(flushableWriter)
+		if !ok {
+			return unknownErrorResponse(fmt.Errorf("Cannot stream history, response handler does not support flushing"))
+		}
+		enc = json.NewEncoder(&streamableResponseWriter{f})
+	} else {
+		enc = json.NewEncoder(w)
+	}
+
 	history := activity.History()
+	if err := writeHistory(r.Context(), enc, history); err != nil {
+		return err
+	}
 
-	w.WriteHeader(http.StatusOK)
+	if follow {
+		last := len(history)
+		for {
+			// Continue sending updates
+			select {
+			case <-r.Context().Done():
```

Comment:
Since we're starting to have multiple streamed responses now (with stream, exec, history, and historyEntry/journal), I think we should extend the streamableResponseWriter to centralize context-cancellation related cleanup (+ the logging related to that).

Reply:
I'm not sure they're directly related, but something to provide a common cancellation pattern makes sense. This starts to conflict with your PR though.

Reply:
I was thinking of something like wrapping streamableResponseWriter#Write to invoke a cleanup function upon context cancellation (and also outputting the log message). That shouldn't conflict with my PR.

Reply:
I don't think that would work. Write is only invoked if more data comes in; it could sit open indefinitely if no more data is received even though the context is cancelled. A separate goroutine waiting on cancellation makes more sense.

Reply:
Ah yeah, you're right. It wouldn't.